contract_line_invoice_by_average/models/contract_line.py

```diff
@@ -0,0 +1,37 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ContractContract(models.Model):
+    _inherit = 'contract.contract'
+
+    @api.multi
+    def _recurring_create_invoice(self, date_ref=False):
+        invoices = super()._recurring_create_invoice(date_ref)
+        for contract_to_invoice in self:
+            product_avg_qty_dict = {}
+            for each_contract_line in contract_to_invoice.contract_line_ids.\
+                    filtered(lambda l: l.products_invoiced_by_avg_ids):
+                for each_avg_product in each_contract_line.products_invoiced_by_avg_ids:
+                    product_avg_qty_dict.update(
+                        {each_avg_product.id: each_contract_line.quantity})
+        for each_invoice in invoices:
```

Comment:
@Swapnesh-SerpentCS I get the following error when running the scheduler manually:

```
[...]
  File "/opt/odoo/src/contract/contract/models/contract.py", line 467, in cron_recurring_create_invoice
    contracts_to_invoice._recurring_create_invoice(date_ref)
  File "/opt/odoo/src/contract/contract_line_invoice_by_average/models/contract_line.py", line 19, in _recurring_create_invoice
    for each_invoice in invoices:
ValueError: <class 'TypeError'>: "'NoneType' object is not iterable" while evaluating
'model.cron_recurring_create_invoice()'
```